ops/internal/deployer/deployer.go

```diff
@@ -0,0 +1,335 @@
+	if _, err := os.Stat(deployerPath); os.IsNotExist(err) {
+		return "", fmt.Errorf("deployer binary not found at path: %s", deployerPath)
+	}
+	return deployerPath, nil
+}
+
+// setupStateAndIntent prepares the deployer environment by creating merged state and intent files
+// in the specified working directory.
+func (d *OpDeployer) setupStateAndIntent(inputStatePath, workdir string) error {
+	// Read the state file
+	state, err := ReadOpaqueMappingFile(inputStatePath)
+	if err != nil {
+		return fmt.Errorf("failed to read state file: %w", err)
+	}
+
+	// Get state merge function based on deployer version
+	mergeFunc, err := getMergeStateFunc(d.DeployerVersion)
+	if err != nil {
+		return fmt.Errorf("failed to determine merge function: %w", err)
+	}
+	mergedIntent, mergedState, err := mergeFunc(state)
+	if err != nil {
+		return fmt.Errorf("failed to merge state: %w", err)
+	}
+
+	// Write state.json in the temp directory
+	stateJSON, err := json.MarshalIndent(mergedState, "", "  ")
+	if err != nil {
+		return fmt.Errorf("failed to marshal state to JSON: %w", err)
+	}
+	stateTempPath := filepath.Join(workdir, "state.json")
+	if err := os.WriteFile(stateTempPath, stateJSON, 0o644); err != nil {
+		return fmt.Errorf("failed to write state to temp file: %w", err)
+	}
+	d.lgr.Info("Wrote state to temporary file", "path", stateTempPath)
+
+	// Write intent.toml in the temp directory
+	useInts(mergedIntent)
+	intentTOML, err := toml.Marshal(mergedIntent)
+	if err != nil {
+		return fmt.Errorf("failed to marshal intent to TOML: %w", err)
+	}
+	intentTempPath := filepath.Join(workdir, "intent.toml")
+	if err := os.WriteFile(intentTempPath, intentTOML, 0o644); err != nil {
+		return fmt.Errorf("failed to write intent to temp file: %w", err)
+	}
+	d.lgr.Info("Wrote intent to temporary file", "path", intentTempPath)
+
+	return nil
+}
+
+// GenerateStandardGenesis runs op-deployer binary to generate a genesis
+// - l1RpcUrl must match the state's L1 and is required by op-deployer, even though we aren't sending any txs
+func (d *OpDeployer) GenerateStandardGenesis(statePath, chainId, l1RpcUrl string) (*OpaqueMapping, error) {
+	workdir, err := d.copyStateFileToTempDir(statePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to copy state file to temporary directory: %w", err)
+	}
+	defer os.RemoveAll(workdir)
+
+	if err := d.setupStateAndIntent(statePath, workdir); err != nil {
+		return nil, fmt.Errorf("failed to setup state and intent: %w", err)
+	}
+
+	deployerPath, err := d.getBinaryPath()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get deployer binary path: %w", err)
+	}
+
+	// We don't want to use a funded account here, because there should not be any txs sent.
+	// All contracts should have already been deployed, and the 'apply' command should skip
+	// those pipeline steps, then only generate the genesis.
+	privateKey, err := ecdsa.GenerateKey(crypto.S256(), rand.Reader)
```

Comment:
Just use the test... junk private key rather than a random one.

Reply:
Done in https://github.com/ethereum-optimism/superchain-registry/pull/1041/commits/cd4a6eb6f4f47948ef16b0470750705218de75f7